### DISCUSSION TITLE
Enable removal of newline following topmost package statement

### DISCUSSION THREAD

**Discussion by andrew-selvia** [scalameta/scalafmt#3629] [Q&A] [answered]
I've attempted to configure scalafmt to remove *all* newlines (including, pertinently, the one added between the topmost package statement and the next statement). Sadly, I've come up short. [Prior discussion](https://github.com/scalameta/scalafmt/pull/1555/) suggests that this is desired broadly. I suspect I'm in the minority.

Nonetheless, I'm curious. Is this already possible? If not, could it be made configurable?

Ideally, this code:
```scala
package org.scalafmt

import scala.collection.Map
```

Could be automatically formatted as follows:
```scala
package org.scalafmt
import scala.collection.Map
```

**Comment by kitbellew** [comment, marked as answer]
if you had removed all newlines, you would have had
```
package org.scalafmtimport scala.collection.Map
```
what you are interested in, we call that _blank lines_, and there's a parameter for that. if it doesn't do what you want, then it's not supported.

**Reply by andrew-selvia** [reply]
Thank you for helping me correct my terminology. Yes, it seems 0 *blank* lines is not possible to achieve with the following configuration:
```hocon
newlines {
  topLevelBodyIfMinStatements = []
  topLevelBodyMinStatements = 0
  topLevelStatementBlankLines = [{ blanks = -1 }]
}
```

I'll convert this discussion into an issue then.
